integreat_cms/summ_ai_api/summ_ai_api_client.py

```diff
@@ -85,21 +92,19 @@ async def translate_text_field(self, session, text_field):
             ) as response:
                 # Wait for the response
                 response_data = await response.json()
-                # Check whether the text was translated successfully
-                if "translated_text" not in response_data:
-                    if "error" in response_data:
-                        raise SummAiException(
-                            f"API error: {response.status} - {response_data['error']}"
-                        )
-                    raise SummAiException(
-                        f"Unexpected API result: {response.status} - {response_data!r}"
-                    )
-                # Let the field handle the translated text
-                text_field.translate(response_data["translated_text"])
-                return text_field
+                if self.check_rate_limit_exceeded(response.status):
+                    # !!! we are not lucky with switching types of return values (text_field when suceeded, string when waiting, none when error)
+                    # what do you suggest?
+                    return self.SIGNAL_WAIT_FOR_NEXT_REQUEST
```

Comment:
Yes! This was also what I was thinking about, but in that case I'd strongly prefer if it would be a subclass of `SummAiException` because it just seems sensible – but then we would need to somehow catch any `SummAiException` *except* if it's a `SummAiRateLimitingExceeded`. I looked into this a bit, and I don't think there is a clean way to do this without simply re-raising the exception.

Reply:
Yes, I agree. I mean you could also use `SummAiException` as base class and define `SummAiRateLimitingExceeded(SummAiException)` and `SummAiRuntimeException(SummAiException)` or whatever as subclasses of it. And then raise a `SummAiRuntimeException` or whatever in all non-rate-limiting cases...

Reply:
Ah, I see! That idea still eluded me, thanks!